python/phylanx/ast/transformation.py

```diff
@@ -392,10 +412,23 @@ def _UnaryOp(self, a, allowreturn=False):
 
     def _For(self, a, allowreturn=False):
         symbol_info = full_node_name(a)
-        ret = "map%s(lambda(" % symbol_info
+        ret = ""
+
+        if is_node(a.iter, 'Call'):
+            func_nom = get_call_func_name(a.iter)
+            if func_nom == 'prange':
+                ret = "parallel_map%s(lambda(" % symbol_info
+            else:
+                ret = "map%s(lambda(" % symbol_info
+        else:          
+            ret = "map%s(lambda(" % symbol_info
```

Comment:
I know this was missing before, but should we annotate the lambda() primitive with symbol information as well?

Reply:
Shouldn't be too hard to add. Will get that added in today.